streamlit_web.py

```diff
@@ -657,23 +662,72 @@ def interactive():
     if ner_cb:
         doc = az.get_nlp(input_text)
         named_entities = az.named_entity_recognization(input_text)
-        if len(named_entities) > 0:
-            html = spacy.displacy.render(doc, style="ent")
-            # Newlines seem to mess with the rendering
-            html = html.replace("\n", " ")
-            HTML_WRAPPER = """<div style="overflow-x: auto; border: 1px solid \
-        #e6e9ef; border-radius: 0.25rem; padding: 1rem;
-        margin-bottom: 2.5rem">\
-        {}</div>"""
-            st.write(HTML_WRAPPER.format(html), unsafe_allow_html=True)
-        else:
-            st.info("No named entity recognized")
+        displacy_renderer(named_entities)
     if sentiment_cb:
         sentiments = TextBlob(az.lemmatized_text(input_text))
         st.write(sentiments.sentiment)
     if summary_cb:
         summaries = sz.summarize_text(input_text)
         st.write(summaries)
 
+
+def entities():
+    """Page to display entity analysis"""
+    st.write("Entity analysis inspects the given text for known entities \
+    and returns information about those entities. It is a way to extract \
+    information that seeks to locate and classify named entities in text \
+    into pre-defined categories such as the names of persons, organizations, \
+    locations, expressions of times, quantities, monetary values, and percentages.")
+
+    # make a copy of the main dataframe
+    input_df = main_df.copy(deep=True)
+    input_df = input_df[input_df[assign_id].isin(assignments)]
+
+    # makes a drop down list to select users classified by assignments
+    for assignment in input_df[assign_id].unique():
+        st.write("")
+        st.subheader(assignment)
+        for student in input_df[stu_id].unique():
+            with st.beta_expander(student):
+                entity_analysis(assignment, student, input_df)
+
+
+def entity_analysis(assignment, student, input_df):
+    """function that selects, modifies and runs the entity analysis on a document"""
+
+    # makes a dataframe with the selected user's information
+    df_selected_stu = input_df.loc[
+        input_df[stu_id].isin([student])
+        & input_df[assign_id].isin([assignment])
+    ]
+
+    # selects the combined column from the dataframe and extracts it
+    combine_start = df_selected_stu.columns.get_loc("combined")
+    combine_end = df_selected_stu.columns.get_loc("combined") + 2
+    df_selected_stu_combined = df_selected_stu.iloc[:,combine_start:combine_end]
```

Comment:
A bug I didn't catch from the last review (or is it newly introduced?): it's displaying both the `normalized` and `combined` column (`normalized` does not have punctuations, so probably not the one to use).